src/features/views/components/ViewColumnDialog/ColumnGallery.tsx

```diff
@@ -12,104 +21,242 @@ import choices, { ColumnChoice } from './choices';
 interface ColumnGalleryProps {
   existingColumns: ZetkinViewColumn[];
   onAdd: (choice: ColumnChoice) => void;
+  onClose: () => void;
   onConfigure: (choice: ColumnChoice) => void;
 }
 
 const ColumnGallery: FunctionComponent<ColumnGalleryProps> = ({
   existingColumns,
   onAdd,
+  onClose,
   onConfigure,
 }) => {
   const intl = useIntl();
   const isMobile = useMediaQuery((theme: Theme) =>
     theme.breakpoints.down('sm')
   );
 
+  const [isSearching, setIsSearching] = useState(false);
+  const [searchString, setSearchString] = useState('');
+  const [searchResults, setSearchResults] = useState<
+    (ColumnChoice | undefined)[]
+  >([]);
```

Comment:
This is _derived state_, which can be the source of many bugs. Could we refactor this to use memoization instead?

Reply:
yes! I don't know use memoization but it sounds good!